src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/DeliverySessionService.kt

```diff
@@ -61,6 +62,84 @@ class DeliverySessionService(
     return deliverySessionRepository.save(session)
   }
 
+  fun scheduleNewDeliverySessionAppointment(
+    referralId: UUID,
+    sessionNumber: Int,
+    appointmentTime: OffsetDateTime,
+    durationInMinutes: Int,
+    updatedBy: AuthUser,
+    appointmentDeliveryType: AppointmentDeliveryType,
+    appointmentSessionType: AppointmentSessionType,
+    appointmentDeliveryAddress: AddressDTO? = null,
+    npsOfficeCode: String? = null,
+  ): DeliverySession {
+    val session = getDeliverySession(referralId, sessionNumber) ?: throw EntityNotFoundException("Session not found for referral [referralId=$referralId, sessionNumber=$sessionNumber]")
+    val existingAppointment = session.currentAppointment
+
+    existingAppointment?.let {
+      if (it.appointmentTime.isAfter(appointmentTime)) {
```

Comment:
Why is this necessary? Isn't it covered by the check that the latest appointment has feedback

Reply:
No because the time given for this new appointment could be before the existing one even though feedback was submitted.